### PR TITLE
[Bug fix] The bug following be fixed:

### PR DESCRIPTION
**Pull request by HappenLee**
1. Fix cast failed in nullable column
2. Fix arithmetic expr core bug
3. Fix core of abs(bigint)
4. Fix union node get wrong type result to wrong answer
5. Fix core bug in math compute in Vec Engine
6. Fix bug of core in union node because of type is wrong
7. Fix bug of mem reuse

## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/columns/column.h [hunk 1]
```diff
@@ -299,10 +299,10 @@ class IColumn : public COW<IColumn> {
         return false;
     }
 
-    MutablePtr mutate(int target_use_count = 1) const&& {
-        MutablePtr res = shallow_mutate(target_use_count);
+    MutablePtr mutate() const&& {
+        MutablePtr res = shallow_mutate();
         res->for_each_subcolumn(
-                [target_use_count](WrappedPtr& subcolumn) { subcolumn = std::move(*subcolumn).mutate(target_use_count + 1); });
+                [](WrappedPtr& subcolumn) { subcolumn = std::move(*subcolumn).mutate(); });
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/vec/columns/column.h:305]: 嗯，我的理解，原则上由父column持有就可以了。但是clickhouse设计的时候是比较复杂的，它支持多个父column持有同一个子column，这会导致进行内存复用等操作的时候需要加倍小心，否则会导致很多诡异的问题。

File: be/src/vec/core/block.h [hunk 2]
```diff
@@ -269,9 +269,19 @@ class MutableBlock {
             }
         } else {
             for (int i = 0; i < _columns.size(); ++i) {
-                _columns[i]->insert_range_from(
+                if (!_data_types[i]->equals(*block.get_by_position(i).type)) {
+                    DCHECK(_data_types[i]->is_nullable());
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/vec/core/block.h:273]: 多个union node合并的时候是常常出现nullable列与非nullable列合并的，但是最终的结果列一定是非nullable列，所以这里有这个判断。

File: be/src/vec/data_types/number_traits.h [hunk 3]
```diff
@@ -63,7 +63,7 @@ struct Construct<false, false, 4> {
 };
 template <>
 struct Construct<false, false, 8> {
-    using Type = UInt64;
+    using Type = Int128;
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/vec/data_types/number_traits.h:66]: UInt64用来表示abs的结果，因为doris不支持，只能用更大int128表示

File: be/src/olap/aggregate_func.h [hunk 4]
```diff
@@ -512,7 +512,7 @@ struct AggregateFuncTraits<OLAP_FIELD_AGGREGATION_BITMAP_UNION, OLAP_FIELD_TYPE_
         // we use zero size represent this slice is a agg object
         dst_slice->size = 0;
         auto bitmap = new BitmapValue(src_slice->data);
-        mem_pool->mem_tracker()->Consume(sizeof(BitmapValue));
+//        mem_pool->mem_tracker()->Consume(sizeof(BitmapValue));
```
[the remark this answers is not in the document]
Comment (HappenLee) [reply be/src/olap/aggregate_func.h:515]: 这个是个失误的注释，这个会导致dcheck不过，我可以后续改回来

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**